app/controllers/prisoners_controller.rb

```diff
@@ -15,6 +15,12 @@ def show
     @tasks = PomTasks.new.for_offender(@prisoner)
     @allocation = Allocation.find_by(nomis_offender_id: @prisoner.offender_no)
 
+    @primary_pom_name = fetch_pom_name(@allocation.primary_pom_nomis_id) if @allocation.present?
```

Comment:
Introducing more instance variables for the two names (`@primary...` and `@secondary...`) suggests to me that we're doing too much in this controller. Is there a way to avoid this?